server/integration/testsuite/src/test/java/org/infinispan/server/test/cs/custom/CustomCacheStoreIT.java

```diff
@@ -48,23 +47,26 @@
     final String cacheLoaderMBean = "jboss.infinispan:type=Cache,name=\"default(local)\",manager=\"local\",component=CacheLoader";
 
     @BeforeClass
-    public static void before() {
-        // need to put the brno loader classes into ispn core jar, so ispn can see them
+    public static void before() throws Exception {
         String serverDir = System.getProperty("server1.dist");
-        File ispnCoreJarDir = new File(serverDir + "/modules/system/layers/base/org/infinispan/main/");
-        FileFilter ispnCoreJarFilter = new WildcardFileFilter("infinispan-core*.jar");
-        File ispnCoreJar = ispnCoreJarDir.listFiles(ispnCoreJarFilter)[0];
 
-        String customLoaderClassesDir = System.getProperty("basedir") + "/target/test-classes/org/infinispan/persistence/cluster/";
-        File[] csClasses = {new File(customLoaderClassesDir + "MyCustomCacheStore.class"),
-                            new File(customLoaderClassesDir + "MyCustomCacheStoreConfiguration.class"),
-                            new File(customLoaderClassesDir + "MyCustomCacheStoreConfigurationBuilder.class")};
-        addFilesToZip(ispnCoreJar, csClasses, "org/infinispan/persistence/cluster/");
+        JavaArchive deployedCacheStore = ShrinkWrap.create(JavaArchive.class);
+        deployedCacheStore.addPackage(MyCustomCacheStore.class.getPackage());
+        deployedCacheStore.addAsManifestResource(new Asset() {
```

Comment:
This can be achieved mu more simply by doing:

`deployedCacheStore .addAsServiceProvider(ExternalStore.class, MyCustomCacheStore.class)`

See `org.infinispan.server.test.client.hotrod.HotRodRemoteCacheIT` for this and other tricks.